packages/lib-classifier/src/components/Classifier/Classifier.js

```diff
@@ -137,10 +143,11 @@ export default function Classifier({
 
   useEffect(function onURLChange() {
     const { workflows } = classifierStore
-    if (loaded && workflowID) {
-      workflows.selectWorkflow(workflowID, subjectSetID, subjectID)
+    if (loaded && workflowData) {
+      const [ workflowSnapshot ] = JSON.parse(workflowData).workflows
+      classifierStore.refreshWorkflow(workflowSnapshot, subjectSetID, subjectID)
     }
-  }, [loaded, subjectID, subjectSetID, workflowID])
+  }, [loaded, subjectID, subjectSetID, workflowData])
```

Comment:
`onURLChange` here runs whenever any part of `workflowData` changes. Maybe we should restrict that to only a subset of workflow properties.